cpp/include/rmm/mr/device/tracking_resource_adaptor.hpp

```diff
@@ -244,10 +244,10 @@ class tracking_resource_adaptor final : public device_memory_resource {
           bytes,
           this->allocations_.size());
       } else {
-        allocations_.erase(found);
-
         auto allocated_bytes = found->second.allocation_size;
```

Comment:
While we're here, let's make this const.
```suggestion
        auto const allocated_bytes = found->second.allocation_size;
```